front/src/applications/stdcm/formatStcmConf.ts

```diff
@@ -1,28 +1,26 @@
 import { Dispatch } from 'redux';
 import { TFunction } from 'i18next';
+import { PathQuery, PostStdcmApiArg } from 'common/api/osrdEditoastApi';
 
-import {
-  ALLOWANCE_UNIT_TYPES,
-  TYPES_UNITS,
-} from 'applications/stdcm/components/OldAllowances/allowancesConsts';
+import { createAllowanceValue } from 'applications/stdcm/components/OldAllowances/allowancesConsts';
 import { STDCM_MODES, OsrdStdcmConfState } from 'applications/operationalStudies/consts';
 import { time2sec } from 'utils/timeManipulation';
 import { makeEnumBooleans } from 'utils/constants';
 
 import { ActionFailure } from 'reducers/main';
 import { ThunkAction } from 'types';
-import { getOpenApiSteps } from 'common/Pathfinding/Pathfinding';
+import { getPathfindingQuery } from 'common/Pathfinding/Pathfinding';
 
 export default function formatStdcmConf(
   dispatch: Dispatch,
   setFailure: (e: Error) => ThunkAction<ActionFailure>,
   t: TFunction,
   osrdconf: OsrdStdcmConfState
-) {
+): PostStdcmApiArg | undefined {
   const { isByOrigin, isByDestination } = makeEnumBooleans(STDCM_MODES, osrdconf.stdcmMode);
 
   let error = false;
-  if (!osrdconf.origin && isByOrigin) {
+  if (!(osrdconf.origin && osrdconf.originUpperBoundTime) && isByOrigin) {
```

Comment:
Why did you add the check on originUpperBoundTime here ? If it is undefined, I'm not sure that you will display the right message

Reply:
that was supposed to be for `originTime`.
both `originTime` and `originUpperBoundTime` are used to get `start_time`